<!DOCTYPE html>
<head>
    <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="">
<meta name="keywords" content=" ">
<title>Syllabus - GEN242 | Data Analysis in Genome Biology</title>
<link rel="stylesheet" href="css/syntax.css">


<link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
<!--<link rel="stylesheet" type="text/css" href="css/bootstrap.min.css">-->
<link rel="stylesheet" href="css/modern-business.css">
<link rel="stylesheet" href="css/lavish-bootstrap.css">
<link rel="stylesheet" href="css/customstyles.css">
<link rel="stylesheet" href="css/theme-blue.css">

<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>
<script src="js/jquery.navgoco.min.js"></script>


<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/2.0.0/anchor.min.js"></script>
<script src="js/toc.js"></script>
<script src="js/customscripts.js"></script>

<link rel="shortcut icon" href="images/favicon.ico">

<!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
<![endif]-->

<link rel="alternate" type="application/rss+xml" title="" href="http://localhost:4000feed.xml">

    <script>
        $(document).ready(function() {
            // Initialize navgoco with default options
            $("#mysidebar").navgoco({
                caretHtml: '',
                accordion: true,
                openClass: 'active', // open
                save: false, // leave false or nav highlighting doesn't work right
                cookie: {
                    name: 'navgoco',
                    expires: false,
                    path: '/'
                },
                slide: {
                    duration: 400,
                    easing: 'swing'
                }
            });

            $("#collapseAll").click(function(e) {
                e.preventDefault();
                $("#mysidebar").navgoco('toggle', false);
            });

            $("#expandAll").click(function(e) {
                e.preventDefault();
                $("#mysidebar").navgoco('toggle', true);
            });

        });

    </script>
    <script>
        $(function () {
            $('[data-toggle="tooltip"]').tooltip()
        })
    </script>
    

</head>
<body>
<!-- Navigation -->
<nav class="navbar navbar-inverse navbar-fixed-top">
    <div class="container topnavlinks">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="fa fa-home fa-lg navbar-brand" href="index.html">&nbsp;<span class="projectTitle"> Data Analysis in Genome Biology</span></a>
        </div>
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <!-- entries without drop-downs appear here -->
                
                
                
                <li><a href="https://piazza.com/class/j10z7wzbqum2tb" target="_blank">Piazza</a></li>
                
                
                
                <li><a href="https://github.com/tgirke/GEN242" target="_blank">GitHub Repo</a></li>
                
                
                
                <!-- entries with drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->
                
                
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Resources<b class="caret"></b></a>
                    <ul class="dropdown-menu">
                        
                        
                        <li><a href="https://cran.r-project.org/" target="_blank">CRAN</a></li>
                        
                        
                        
                        <li><a href="http://bioconductor.org/" target="_blank">Bioconductor</a></li>
                        
                        
                        
                        <li><a href="http://manuals.bioinformatics.ucr.edu/home/R_BioCondManual" target="_blank">R & Bioconductor Manual</a></li>
                        
                        
                        
                        <li><a href="https://guides.github.com/activities/hello-world/" target="_blank">GitHub</a></li>
                        
                        
                        
                        <li><a href="http://hpcc.ucr.edu/" target="_blank">HPC Center (HPCC)</a></li>
                        
                        
                        
                        <li><a href="http://girke.bioinformatics.ucr.edu/" target="_blank">Girke Lab</a></li>
                        
                        
                        
                        <li><a href="http://idratherbewriting.com/documentation-theme-jekyll" target="_blank">Jekyll Doc Theme</a></li>
                        
                        
                    </ul>
                </li>
                
                
                
			<li>



  <a class="email" title="Submit feedback" href="#" onclick="javascript:window.location='mailto:thomas.girke@ucr.edu?subject=GEN242 Class Web Site feedback&body=I have some feedback about the Syllabus - GEN242 page: ' + window.location.href;"><i class="fa fa-envelope-o"></i> Feedback</a>

</li>

		
                <!--comment out this block if you want to hide search-->
                <li>
                    <!--start search-->
                    <div id="search-demo-container">
                        <input type="text" id="search-input" placeholder="search...">
                        <ul id="results-container"></ul>
                    </div>
                    <script src="js/jekyll-search.js" type="text/javascript"></script>
                    <script type="text/javascript">
                            SimpleJekyllSearch.init({
                                searchInput: document.getElementById('search-input'),
                                resultsContainer: document.getElementById('results-container'),
                                dataSource: 'search.json',
                                searchResultTemplate: '<li><a href="{url}" title="Syllabus - GEN242">{title}</a></li>',
                    noResultsText: 'No results found.',
                            limit: 10,
                            fuzzy: true,
                    })
                    </script>
                    <!--end search-->
                </li>
            </ul>
        </div>
        </div>
        <!-- /.container -->
</nav>

<!-- Page Content -->
<div class="container">
    <div class="col-lg-12">&nbsp;</div>
    <!-- Content Row -->
    <div class="row">
        <!-- Sidebar Column -->
        <div class="col-md-3">

          












<ul id="mysidebar" class="nav">
    <li class="sidebarTitle"><center><font color="#26466D" size="6">GEN242</font></center> <center><font color="#8b9299" size="3">Spring 2017</font></center></li>
    
    
    
        
    
    <li>
        <a href="#">Home</a>
        <ul>
            
            
            
            <li><a href="index.html">Introduction</a></li>
            
            
            
            
            
            
            <li class="active"><a href="mydoc_syllabus.html">Syllabus</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Internal</a>
        <ul>
            
            
            
            <li><a href="mydoc_internal.html">Internal Resources</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Schedule</a>
        <ul>
            
            
            
            <li><a href="mydoc_schedule.html">Course Schedule</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Lecture Slides</a>
        <ul>
            
            
            
            <li><a href="mydoc_slides_01.html">Course Introduction</a></li>
            
            
            
            
            
            
            <li><a href="mydoc_slides_02.html">Molbiol Basics</a></li>
            
            
            
            
            
            
            <li><a href="mydoc_slides_03.html">Databases and Software</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Homework</a>
        <ul>
            
            
            
            <li><a href="mydoc_homework_01.html">HW1 - Online Tools</a></li>
            
            
            
            
            
            
            <li><a href="mydoc_homework_02.html">HW2 - Biocluster/Linux</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">GitHub</a>
        <ul>
            
            
            
            <li><a href="mydoc_tutorial_01.html">GitHub</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Biocluster & Linux</a>
        <ul>
            
            
            
            <li><a href="mydoc_tutorial_02.html">Linux Basics</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Introduction to R</a>
        <ul>
            
        </ul>
        
        
    
    <li>
        <a href="#">Programming in R</a>
        <ul>
            
        </ul>
        
        
    
    <li>
        <a href="#">NGS Analysis Basics</a>
        <ul>
            
        </ul>
        
        
    
    <li>
        <a href="#">NGS Workflow Overview</a>
        <ul>
            
        </ul>
        
        
    
    <li>
        <a href="#">RNA-Seq Workflow</a>
        <ul>
            
        </ul>
        
        
    
    <li>
        <a href="#">ChIP-Seq Workflow</a>
        <ul>
            
        </ul>
        
        
    
    <li>
        <a href="#">VAR-Seq Workflow</a>
        <ul>
            
        </ul>
        
        
    
    <li>
        <a href="#">Graphics and Visualization</a>
        <ul>
            
        </ul>
        
        
    
    <li>
        <a href="#">Cluster Analysis</a>
        <ul>
            
        </ul>
        
        
    
    <li>
        <a href="#">Paper Presentations</a>
        <ul>
            
        </ul>
        
        
    
    <li>
        <a href="#">Course Projects</a>
        <ul>
            
        </ul>
        
        
        
        <!-- if you aren't using the accordion, uncomment this block:
           <p class="external">
               <a href="#" id="collapseAll">Collapse All</a> | <a href="#" id="expandAll">Expand All</a>
           </p>
           -->
    </li>
</ul>
</div>

<!-- this highlights the active parent class in the navgoco sidebar. this is critical so that the parent expands when you're viewing a page. This must appear below the sidebar code above. Otherwise, if placed inside customscripts.js, the script runs before the sidebar code runs and the class never gets inserted.-->
<script>$("li.active").parents('li').toggleClass("active");</script>

    <!-- Content Column -->
    <div class="col-md-9">
        <div class="post-header">
   <h1 class="post-title-main">Syllabus - GEN242</h1>
</div>



<div class="post-content">

   

    
    
<!-- this handles the automatic toc. use ## for subheads to auto-generate the on-page minitoc. if you use html tags, you must supply an ID for the heading element in order for it to appear in the minitoc. -->
<script>
$( document ).ready(function() {
  // Handler for .ready() called.

$('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2,h3,h4' });

/* this offset helps account for the space taken up by the floating toolbar. */
$('#toc').on('click', 'a', function() {
  var target = $(this.getAttribute('href'))
    , scroll_target = target.offset().top

  $(window).scrollTop(scroll_target - 10);
  return false
})
  
});
</script>

<div id="toc"></div>

    


<!--
    

    <a target="_blank" href="https://github.com/tgirke/GEN242/blob/gh-pages/pages//mydoc_syllabus.md" class="btn btn-default githubEditButton" role="button"><i class="fa fa-github fa-lg"></i> Edit me</a>
-->
    

  <h2 id="course-title">Course title</h2>

<p>Data Analysis in Genome Biology <br />
GEN242 - Spring 2017</p>

<h2 id="instructor">Instructor</h2>
<p>Name: Thomas Girke <br /> 
Email: thomas.girke@ucr.edu <br />
Office location: 1207F Genomics <br />
Office hours: Tue &amp; Thu 4:30 - 5:30 PM</p>

<h2 id="description">Description</h2>

<p>Introduction to algorithms, statistical methods and data analysis programming
routines relevant for genome biology. The class consists of three
main components: lectures, hands-on practicals and student course projects. The
lecture topics cover databases, sequence (NGS) analysis, phylogenetics,
comparative genomics, genome-wide profiling methods, network biology and more.
The hands-on practicals include homework assignments and course projects
focusing on data analysis programming of next generation genome data using
command-line tools on a computer cluster and the programming environment R.
Credit: 4 units (4 hours lecture, 2 hours discussion)</p>

<h2 id="objectives-of-course">Objectives of course</h2>

<ul>
  <li>Acquire understanding of algorithms used in bioinformatics</li>
  <li>Obtain hands-on experience in large scale data analysis.</li>
</ul>

<h2 id="prerequisites">Prerequisites</h2>
<p>The main prerequisite for this course is a strong interest in acquiring the
skills required for mastering the computational aspects of modern genome
research.</p>

<h2 id="structure-of-course">Structure of course</h2>
<p>Two lectures per week (2 hours each) plus one discussion section (1 hour).
During the first weeks the discussion section will be used for data analysis
tutorials using Linux command-line tools and R.</p>

<h2 id="time">Time</h2>
<p>Lecture: Tue/Thu 2:10-3:30 PM <br />
Discussion: Thu 3:40-4:30 PM</p>

<h2 id="location">Location</h2>
<p><a href="http://classrooms.cmsdev.ucr.edu/buildings/chassINTS/2130.html">2130 CHASS Interdisciplinary Bldg-South (INST)</a></p>

<h2 id="grading">Grading</h2>
<ol>
  <li>Homework assignments: 40%</li>
  <li>Scientific paper presentation: 20%</li>
  <li>Course project presentations: 20%</li>
  <li>Final project report: 20%</li>
</ol>

<h2 id="materials-needed">Materials needed</h2>
<p>Students are expected to bring to each class meeting a laptop with a functional wireless
connection and a recent internet browser version (e.g. Firefox, Chrome or
Safari) preinstalled. Tablet computers with mobile operating systems are not
suitable for running the required software. User accounts on a research
computer cluster will be provided at the beginning of the course. To log in to
the cluster, students also need to install a terminal application for their
operating system (<em>e.g.</em> <a href="http://www.iterm2.com/#/section/home">iTerm2</a> on OS X,
and <a href="http://www.chiark.greenend.org.uk/~sgtatham/putty/download.html">PuTTY</a> or 
<a href="http://mobaxterm.mobatek.net/">MobaXterm</a> on Windows) as well as a file exchange software such as
<a href="https://filezilla-project.org/download.php?type=client">FileZilla</a>. In
addition, a recent version of <a href="http://www.r-project.org">R</a> and
<a href="http://rstudio.org">RStudio</a> should be installed on each laptop.</p>

<h2 id="schedule">Schedule</h2>

<table>
  <thead>
    <tr>
      <th>Week</th>
      <th>Topic</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Week 1</td>
      <td>Course Introduction</td>
    </tr>
    <tr>
      <td> </td>
      <td>Databases and Software for Genome Biology</td>
    </tr>
    <tr>
      <td> </td>
      <td>Tutorial: Introduction to Linux and Biocluster</td>
    </tr>
    <tr>
      <td>Week 2</td>
      <td>Sequencing Technologies</td>
    </tr>
    <tr>
      <td> </td>
      <td>Tutorial: Introduction to R</td>
    </tr>
    <tr>
      <td>Week 3</td>
      <td>Sequence Alignments and Searching</td>
    </tr>
    <tr>
      <td> </td>
      <td>Tutorial: Programming in R</td>
    </tr>
    <tr>
      <td>Week 4</td>
      <td>Multiple Sequence Alignments</td>
    </tr>
    <tr>
      <td> </td>
      <td>Short Read Alignment Algorithms</td>
    </tr>
    <tr>
      <td> </td>
      <td>Tutorial: Basics of NGS Analysis</td>
    </tr>
    <tr>
      <td>Week 5</td>
      <td>Gene Expression Analysis using Microarrays and RNA-Seqs</td>
    </tr>
    <tr>
      <td> </td>
      <td>Tutorial: NGS Workflow Overview</td>
    </tr>
    <tr>
      <td> </td>
      <td>Tutorial: RNA-Seq Analysis</td>
    </tr>
    <tr>
      <td>Week 6</td>
      <td>Analysis of ChIP-Seq and VAR-Seq Experiments</td>
    </tr>
    <tr>
      <td> </td>
      <td>Tutorial: ChIP-Seq Analysis</td>
    </tr>
    <tr>
      <td> </td>
      <td>Tutorial: VAR-Seq Analysis</td>
    </tr>
    <tr>
      <td>Week 7</td>
      <td>Student Paper Presentations</td>
    </tr>
    <tr>
      <td>Week 8</td>
      <td>Clustering algorithms</td>
    </tr>
    <tr>
      <td> </td>
      <td>Annotation Systems and Gene Set Enrichment Analysis</td>
    </tr>
    <tr>
      <td> </td>
      <td>Tutorial: Gene Set Enrichment Analysis</td>
    </tr>
    <tr>
      <td>Week 9</td>
      <td>Profile HMMs for Protein Family Modeling</td>
    </tr>
    <tr>
      <td> </td>
      <td>Phylogenetics</td>
    </tr>
    <tr>
      <td> </td>
      <td>Tutorial: Graphics and Data Visualization</td>
    </tr>
    <tr>
      <td>Week 10</td>
      <td>Student Project Presentations</td>
    </tr>
    <tr>
      <td> </td>
      <td>Final Course Discussion</td>
    </tr>
  </tbody>
</table>

<h2 id="reading-list">Reading list</h2>

<h3 id="journal-articles">Journal articles</h3>

<p>Alkan, C, Sajjadian, S, Eichler, E E (2011) Limitations of next-generation genome sequence assembly. Nat Methods, 8: 61-65. http://www.hubmed.org/display.cgi?uids=21102452</p>

<p>Anders, S, Reyes, A, Huber, W (2012) Detecting differential usage of exons from RNA-seq data. Genome Res, 22: 2008-2017. http://www.hubmed.org/display.cgi?uids=22722343</p>

<p>DePristo, M A, Banks, E, Poplin, R, Garimella, K V, Maguire, J R, Hartl, C, Philippakis, A A, del Angel, G, Rivas, M A, Hanna, M, McKenna, A, Fennell, T J, Kernytsky, A M, Sivachenko, A Y, Cibulskis, K, Gabriel, S B, Altshuler, D, Daly, M J (2011) A framework for variation discovery and genotyping using next-generation DNA sequencing data. Nat Genet, 43: 491-498.
http://www.hubmed.org/display.cgi?uids=21478889</p>

<p>Dobin, A., Davis, C.A., Schlesinger, F., Drenkow, J., Zaleski, C., Jha, S., Batut, P., Chaisson, M., Gingeras, T.R., 2012. STAR: ultrafast universal RNA-seq aligner. Bioinformatics 29, 15–21.
http://www.hubmed.org/display.cgi?uids=23104886</p>

<p>Grabherr, M G, Haas, B J, Yassour, M, Levin, J Z, Thompson, D A, Amit, I, Adiconis, X, Fan, L, Raychowdhury, R, Zeng, Q, Chen, Z, Mauceli, E, Hacohen, N, Gnirke, A, Rhind, N, di Palma, F, Birren, B W, Nusbaum, C, Lindblad-Toh, K, Friedman, N, Regev, A (2011) Full-length transcriptome assembly from RNA-Seq data without a reference genome. Nat Biotechnol, 29: 644-652.
http://www.hubmed.org/display.cgi?uids=21572440</p>

<p>Langmead, B, Salzberg, S L (2012) Fast gapped-read alignment with Bowtie 2. Nat Methods, 9: 357-359. http://www.hubmed.org/display.cgi?uids=22388286</p>

<p>Landt et al. (2012) ChIP-seq guidelines and practices of the ENCODE and modENCODE consortia. Genome Res, 22: 1813-1831. http://www.hubmed.org/display.cgi?uids=22955991</p>

<p>Li, H, Durbin, R (2009) Fast and accurate short read alignment with Burrows-Wheeler transform. Bioinformatics, 25: 1754-1760. http://www.hubmed.org/display.cgi?uids=19451168</p>

<p>Machanick, P, Bailey, T L (2011) MEME-ChIP: motif analysis of large DNA datasets. Bioinformatics, 27: 1696-1697.http://www.hubmed.org/display.cgi?uids=21486936</p>

<p>Soneson, C, Delorenzi, M (2013) A comparison of methods for differential expression analysis of RNA-seq data. BMC Bioinformatics, 14: 91-91. http://www.hubmed.org/display.cgi?uids=23497356</p>

<p>Tompa, M, Li, N, Bailey, T L, Church, G M, De Moor, B, Eskin, E, Favorov, A V, Frith, M C, Fu, Y, Kent, W J, Makeev, V J, Mironov, A A, Noble, W S, Pavesi, G, Pesole, G, R{'e}gnier, M, Simonis, N, Sinha, S, Thijs, G, van Helden, J, Vandenbogaert, M, Weng, Z, Workman, C, Ye, C, Zhu, Z (2005) Assessing computational tools for the discovery of transcription factor binding sites. Nat Biotechnol, 23: 137-144.
http://www.hubmed.org/display.cgi?uids=15637633</p>

<p>Trapnell, C, Hendrickson, D G, Sauvageau, M, Goff, L, Rinn, J L, Pachter, L (2013) Differential analysis of gene regulation at transcript resolution with RNA-seq. Nat Biotechnol, 31: 46-53. 
http://www.hubmed.org/display.cgi?uids=23222703</p>

<p>Wilbanks, E G, Facciotti, M T (2010) Evaluation of algorithm performance in ChIP-seq peak detection. PLoS One, 5. http://www.hubmed.org/display.cgi?uids=20628599</p>

<p>Zeitouni, B, Boeva, V, Janoueix-Lerosey, I, Loeillet, S, Legoix-n{'e}, P, Nicolas, A, Delattre, O, Barillot, E (2010) SVDetect: a tool to identify genomic structural variations from paired-end and mate-pair sequencing data. Bioinformatics, 26: 1895-1896. http://www.hubmed.org/display.cgi?uids=20639544</p>

<h3 id="books">Books</h3>

<p>Note: there is no need to purchase any books for this course as most reading material will be based on journal articles!</p>

<p>General
Jonathan Pevsner (2009) Bioinformatics and Functional Genomics. Wiley-Blackwell; 2nd Edition, 992 pages.</p>

<p>Algorithms
Jones N and Pevzner P (2004) An Introduction to Bioinformatics Algorithms. MIT Press, Massachusetts, 435 pages.</p>

<p>Sequence Analysis
Durbin, R, Eddy, S, Krogh, A, Mitchison, G. (1998) Biological Sequence Analysis: Probabilistic Models of Proteins and Nucleic Acids. Cambridge University Press, UK, 356 pages.</p>

<p>Parida L (2008) Pattern Discovery in Bioinformatics: Theory &amp; Algorithms. CRC Press, London, 526 pages.</p>

<p>Profiling Bioinformatics
Gentleman, R, Carey, V, Dudoit, S, Irizarry, R, Huber, W (2005) Bioinformatics and Computational Biology Solutions Using R and Bioconductor. Springer, New York, 473 pages.</p>

<p>Phylogenetics 
Felsenstein, J (2004) Inferring Phylogenies. Sinauer, Massachusetts, 664 pages.</p>

<p>Paradis (2006) Analysis of Phylogenetics and Evolution with R. Springer, New York, 211 pages.</p>



    <div class="tags">
        
    </div>

    

</div>

<hr class="shaded"/>

<footer>
            <div class="row">
                <div class="col-lg-12 footer">
               &copy;2017 UC Riverside. All rights reserved. <br />
 Site last generated: Apr 5, 2017 <br />
<!--
<p><img src="images/company_logo.png" alt="Company logo"/></p>
-->
</div>
            </div>
</footer>


    </div>
    <!-- /.row -->
</div>
<!-- /.container -->
    </div>

</body>

</html>